eo-maven-plugin/src/test/java/org/eolang/maven/ShakeMojoTest.java

```diff
@@ -0,0 +1,234 @@
+            .execute(ShakeMojo.class);
+        MatcherAssert.assertThat(
+            new XMLDocument(
+                new HmBase(temp).load(
+                    Paths.get(
+                        String.format(
+                            this.key,
+                            ShakeMojo.DIR,
+                            TranspileMojo.EXT
+                        )
+                    )
+                ).asBytes()
+            ),
+            Matchers.is(new XMLDocument(cached.asString()))
+        );
+    }
+
+    @Test
+    void skipsAlreadyShaken(@TempDir final Path temp) throws IOException {
+        final FakeMaven maven = new FakeMaven(temp)
+            .withHelloWorld()
+            .execute(ParseMojo.class)
+            .execute(OptimizeMojo.class)
+            .execute(ShakeMojo.class);
+        final Path path = maven.result().get(
+            String.format(this.key, ShakeMojo.DIR, TranspileMojo.EXT)
+        );
+        final long mtime = path.toFile().lastModified();
+        maven.execute(ShakeMojo.class);
+        MatcherAssert.assertThat(
+            path.toFile().lastModified(),
+            Matchers.is(mtime)
+        );
+    }
+
+    @Test
+    void shakesIfExpired(@TempDir final Path temp) throws Exception {
+        final FakeMaven maven = new FakeMaven(temp);
+        final Path tgt = maven
+            .withHelloWorld()
+            .execute(ParseMojo.class)
+            .execute(OptimizeMojo.class)
+            .execute(ShakeMojo.class)
+            .result()
+            .get(
+                String.format(this.key, ShakeMojo.DIR, TranspileMojo.EXT)
+            );
+        final long old = System.currentTimeMillis() - TimeUnit.DAYS.toMillis(10L);
+        if (!tgt.toFile().setLastModified(old)) {
+            Assertions.fail(String.format("The last modified attribute can't be set for %s", tgt));
+        }
+        maven.execute(ShakeMojo.class);
+        MatcherAssert.assertThat(
+            "We expect that already shaken xmir will be replaced by a new shaken xmir, because the first xmir is outdated and should be updated",
+            tgt.toFile().lastModified(),
+            Matchers.greaterThan(old)
+        );
+    }
+
+    @Test
+    void savesShakenResultsToCache(@TempDir final Path temp) throws IOException {
+        final Path cache = temp.resolve("cache");
+        final String hash = "abcdef1";
+        new FakeMaven(temp)
+            .withHelloWorld()
+            .with("cache", cache)
+            .allTojosWithHash(() -> hash)
+            .execute(ParseMojo.class)
+            .execute(OptimizeMojo.class)
+            .execute(ShakeMojo.class);
+        MatcherAssert.assertThat(
+            cache.resolve(ShakeMojo.SHAKEN)
+                .resolve(hash)
+                .resolve("foo/x/main.xmir").toFile(),
+            FileMatchers.anExistingFile()
+        );
+    }
+
+    /**
+     * The test with high number of eo programs reveals concurrency problems of the OptimizeMojo.
```

Comment:
@Yanich96 I think `OptimizeMojo` in the comment is not relevant to the test

Reply:
@maxonfjvipon Should I delete this comment?

Reply:
@Yanich96 I believe you just copied that comment from `OptimizeMojoTest`, but didn't replace "problems of the OptimizeMojo" with "problems of the ShakeMojo". If you think that the comment does not belong here - delete it, otherwise I don't understand why `OptimizeMojo` is mentioned here